spinalcordtoolbox/scripts/sct_deepseg.py

```diff
@@ -314,6 +320,29 @@ def main(argv: Sequence[str]):
         # Use the result of the current model as additional input of the next model
         fname_prior = fname_seg
 
+        # write JSON sidecar file
+        sidecar_json = {
+            'GeneratedBy': [
+                {
+                    'Name': 'spinalcordtoolbox',
+                    "Version": __version__,
+                    "CodeURL": "https://github.com/spinalcordtoolbox/spinalcordtoolbox"
+                }
+            ]
+        }
+        source_path = os.path.join(path_model, "source.json")
+        if os.path.isfile(source_path):
+            with open(source_path, "r") as fp:
+                source_dict = json.load(fp)
+        sidecar_json['GeneratedBy'].append(
+            {
+                'Name': f'sct_deepseg -task {arguments.task}',
+                "Version": source_dict["model_urls"][name_model][0]
+            }
+        )
```

Comment:
This shouldn't be a second `GeneratedBy` entry. Instead, the BIDS spec allows custom fields, so we can differentiate between `CodeURL` vs. `ModelURL`, and `SCTVersion` vs. `ModelVersion`.